File: abc302/b/main.cpp
// "Copyright 2022 tam"
#include <bits/stdc++.h>

using namespace std;

#define rep(i, n) for (int i = 0; i < n; i++)
#define Rep(i, n) for (int i = 1; i <= n; i++)
#define ALL(v) v.begin(), v.end()
#define chmax(x, y) x = max(x, y)
#define chmin(x, y) x = min(x, y)

#define ll int64_t

using P = pair<int, int>;

const int MOD = 1000000007;

int main() {
    int h, w;
    cin >> h >> w;
    vector<string> grid(h);
    for (auto &g : grid) cin >> g;

    const string base = "snuke";
    vector<int> d = {-1, 0, 1};
    rep(i, h) rep(j, w) {
        for (int dh : d) {
            for (int dw : d) {
                vector<pair<int, int>> ans;
                rep(k, 5) {
                    int ch = i + dh * k, cw = j + dw * k;
                    if (ch < 0 || h <= ch || cw < 0 || w <= cw) {
                        continue;
                    }
                    if (grid[ch][cw] != base[k]) break;
                    ans.emplace_back(ch, cw);
                }
                if (ans.size() == 5) {
                    for (auto [ch, cw] : ans) {
                        cout << ch + 1 << " " << cw + 1 << endl;
                    }
                    return 0;
                }
            }
        }
    }

    return 0;
}

File: abc302/b/main.go
package main

import "fmt"

func main() {
	var h, w int
	fmt.Scanf("%d %d", &h, &w)
	var grid []string
	for i := 0 ; i < h ; i++ {
		var s string
		fmt.Scan(&s)
		grid = append(grid, s)
	}

	base := "snuke"

	for i := 0 ; i < h ; i++ {
		for j := 0 ; j < w ; j++ {
			for _, yd := range []int{-1, 1, 0} {
				for _, xd := range []int{-1, 1, 0} {
					cnt := make([][]int, 0, 0)
					for d := 0 ; d < 5 ; d++ {
						cy, cx := i + d * yd, j + d * xd
						if 0 <= cy && cy < h && 0 <= cx && cx < w {
							if grid[cy][cx] != base[d] {
								break;
							}
							cnt = append(cnt, []int{cy + 1, cx + 1})
						}
					}
					if len(cnt) == 5 {
						for _, cc := range cnt {
							fmt.Printf("%d %d\n", cc[0], cc[1])
						}
						return
					}
				}
			}
		}
	}
}